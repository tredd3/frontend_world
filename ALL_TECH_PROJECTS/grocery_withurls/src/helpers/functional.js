export const pipe = (...args) => x => args.reduce((acc, fn) => fn(acc), x);

export const range = length => [...Array(length).keys()];

export const pad = (length, padCharacter) => value => {
  if (String(value).length >= length) return String(value);
  return range(length - String(value).length).map(() => padCharacter).join('') + String(value);
};

const make2digit = pad(2, '0');

export const formatDate = dateLike => {
  const date = new Date(dateLike);
  return `${date.getFullYear()}-${make2digit(date.getMonth() + 1)}-${make2digit(date.getDate())}`;
};

export const objectToQueryString = obj => Object.entries(obj)
  .reduce((acc, [key, value]) => [...acc, `${encodeURIComponent(key)}=${encodeURIComponent(Array.isArray(value)
    ? value.join(',') : value)}`], []).join('&');

export const queryStringToObject = qs => {
  if (!qs) return {};
  return qs.split('&')
    .map(aQsPart => aQsPart.split('=').map(decodeURIComponent))
    .reduce((acc, [name, value]) => {
      const convertedValue = value.includes(',') ? value.split(',') : [value];
      return { ...acc, [name]: convertedValue };
    }, {});
};
