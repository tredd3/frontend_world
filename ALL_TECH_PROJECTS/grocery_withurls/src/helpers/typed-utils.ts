export const flatMap = <T, U>(arr: T[], fn: ((x: T) => U[])): U[] => (
  arr.reduce(
    (acc: U[], x: T) => [...acc, ...fn(x)],
    [] as U[]
  )
);
