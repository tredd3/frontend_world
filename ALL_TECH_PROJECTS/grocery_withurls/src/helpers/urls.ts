export const accountUrl = '/account';
export const cartUrl = '/cart';
export const notificationsUrl = '/notifications';
export const productUrl = (skuId: number | string) => `/products/${skuId}`;
export const productSimilarItemsUrl = (skuId: number | string) => `/products/${skuId}/similar-items`;
export const orderPlacedUrl = (orderId: number | string) => `/order-placed/${orderId}`;
export const ordersUrl = '/orders';
export const orderSearchUrl = '/orders/search';
export const orderDetailsUrl = (orderId: number | string) => `/orders/${orderId}`;
export const wishlistUrl = '/wishlist';
export const categoriesUrl = '/categories';
export const categoryUrl = (categoryId: number | string) => `/categories/${categoryId}`;
export const subCategoryUrl = (categoryId: number | string, subCategoryId: number | string) => `/categories/${categoryId}/${subCategoryId}`;
export const searchUrl = '/search';
export const searchResultsUrl = (searchText: string) => `/search/${searchText}`;
export const collectionUrl = (collectionId: number | string, type?: number | string, subtype?: number | string) => {
  let queryParams;
  if (type) {
    queryParams = new URLSearchParams();
    queryParams.append('type', String(type));

    if (subtype) {
      queryParams.append('subtype', String(subtype));
    }
  }
  return `/collections/${collectionId}${queryParams ? `?${queryParams.toString()}` : ''}`;
};
