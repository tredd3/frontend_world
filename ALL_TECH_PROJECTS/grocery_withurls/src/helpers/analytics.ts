import { getMobileOperatingSystem } from './utilities';
import { getUser } from '../services/user';
import { getCart, PaymentMethodId, paymentLabelById } from '../services/cart';

import { CheckoutData } from '../services/checkout';
import { CartPivot } from '../types';
import { flatMap } from './typed-utils';

let purchaseJourney: string;

export const setPurchaseJourney = (val: string) => {
  purchaseJourney = val;
};

export const getPurchaseJourney = () => purchaseJourney;

declare global {
  // eslint-disable-next-line @typescript-eslint/consistent-type-definitions
  interface Window {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    digitalData: Record<string, any>;
    _satellite: {
      track: (event: string) => void;
    };
  }
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const track = async (event: string, data: Record<string, any>) => {
  if (typeof window === 'undefined') return;
  const { userId } = await getUser();
  const userOS = getMobileOperatingSystem();
  const platform = userOS === 2 ? 'Android App' : userOS === 3 ? 'iOS App' : 'Web';

  const digitalData = {
    link: {
      ...data.link
    },
    internal: {
      ...data.internal
    },
    page: {
      pageInfo: {
        appName: 'JioMart Web'
      },
      ...data.page
    },
    user: {
      platform,
      loginStatus: 'Logged in',
      martId: userId,
      jioCenter: '',
      ...data.user
    },
    product: [
      ...(data.product ? data.product : [])
    ],
    interPromotion: {
      ...data.interPromotion
    },
    ...(data.product ? data.filterSection : {}),
    ...(data.purchase ? data.purchase : {})
  };

  window.digitalData = {
    ...window.digitalData,
    ...digitalData
  };

  // eslint-disable-next-line no-underscore-dangle
  if (window._satellite) window._satellite.track(event);
};

type LinkPositions = 'Top' | 'Middle' | 'Bottom';

export const trackLink = (linkName: string, linkType: string, linkPosition: LinkPositions) => (
  track('linkTracking', { link: { linkName, linkType, linkPosition } })
);

export const trackPage = (pageName: string) => (
  track('pageLoad', { page: { pageName } })
);

type FilterSection = {
  filterSelection: boolean;
  sortSelection: boolean;
}
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const trackProduct = (eventName: string, productData: Record<string, any>, filterSection?: FilterSection) => {
  const defaultFilterSection = {
    filterSelection: false,
    sortSelection: false
  };
  return track(eventName, {
    product: [productData],
    filterSection: (filterSection || defaultFilterSection)
  });
};

export const trackAPIShipments = (eventName: string, shipments: CheckoutData['Shipments'], purchase?: any) => {
  if (shipments && shipments.length) {
    track(eventName, {
      product: flatMap(shipments, shipment => shipment.Product)
        .map(product => ({
          productID: product.ProductSkuId,
          quantity: product.SelectedQuantity,
          price: product.SelectedQuantity * product.SP,
          productName: product.ProductName
        })),
      ...(purchase && { purchase })
    });
  }
};

type PurchaseArg = {
  paymentMethod: PaymentMethodId;
  paymentStatus?: 'Paid' | 'Unpaid';
  cart: CartPivot;
  failureReason?: string;
}

export const trackShipments = async (eventName: string, purchase?: PurchaseArg) => {
  const { shipments } = purchase?.cart || await getCart();
  if (!shipments || !shipments.length) return Promise.resolve();

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  let additionalInfo: Record<string, any> | null = null;

  if (purchase) {
    const { paymentMethod, cart, ...rest } = purchase;
    additionalInfo = {
      paymentMethod: paymentLabelById(paymentMethod),
      couponDiscount: cart.couponDiscountAmount,
      deliveryCharges: cart.deliveryCharge,
      totalOrderValue: cart.totalPayableAmount,
      transactionId: cart.cartId,
      ...rest
    };
  }

  return track(eventName, {
    product: flatMap(shipments, shipment => shipment.products)
      .map(product => ({
        productId: product.skuId,
        quantity: product.selectedQuantity,
        price: product.selectedQuantity * product.sp,
        productName: product.name
      })),
    ...(additionalInfo && additionalInfo)
  });
};
