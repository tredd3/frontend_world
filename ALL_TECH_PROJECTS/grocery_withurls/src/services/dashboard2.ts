import promiseMemoize from 'promise-memoize';
import { APIDashboardProduct } from '../types/api';
import makeApiCall from './helpers/make-api-call';
import { apiRoutes } from './api-routes';
import { DashboardProductPivot } from '../types';
import { pivotDashboardProduct } from './helpers/pivots';
import { getUser } from './user';

type Banner = { id: number; imageUrl: string };
type BannerWithTitle = Banner & { title: string };
type WidgetOfType<T> = { type: T; subType: number };
type BannerWidget<T> = WidgetOfType<T> & { banners: Banner[] };

export type HorizontalScrollableIcons = WidgetOfType<1> & { banners: BannerWithTitle[] };
export type Pager = BannerWidget<3>;
export type ThinCarouselBanner = BannerWidget<4>;
export type BigSquareCarouselBanner = BannerWidget<5>;
export type SmallSquareCarouselBanner = BannerWidget<6> & { title: string }
export type HorizontalProducts = WidgetOfType<7> & {
  title: string;
  viewAllLabel: string;
  products: DashboardProductPivot[];
};
export type W4x = WidgetOfType<11> & { banners: BannerWithTitle[] };
export type W9x = WidgetOfType<12> & { banners: BannerWithTitle[] } & { title: string };
export type W4xWithBanner = BannerWidget<15> & { title: string };
export type W9xWithBanner = BannerWidget<16> & { title: string };

export type Widget =
  HorizontalScrollableIcons
  | Pager
  | ThinCarouselBanner
  | BigSquareCarouselBanner
  | SmallSquareCarouselBanner
  | HorizontalProducts
  | W4x
  | W9x
  | W4xWithBanner
  | W9xWithBanner

type DashboardRequestType = {
  pincode?: number;
  storeId?: number;
};

type APIBanner = {
  bannerId: number;
  imageUrl: string;
};

type APIMiscellaneous = {
  name: string;
  mscId: number;
  imageUrl: string;
}

type APIWidget = {
  type: Widget['type'];
  subType: number;
  title: string;
  deepLinkText: string;
  data: {
    banners: null | APIBanner[];
    products: null | APIDashboardProduct[];
    miscellaneous: null | APIMiscellaneous[];
  };
};

export type DashboardResponseType = {
  widgets: APIWidget[];
};

const widgetBase = (widget: APIWidget) => ({
  type: widget.type,
  subType: widget.subType
});

const apiBannerToBanner = (banner: APIBanner | APIMiscellaneous): Banner => ({
  id: (banner as APIBanner).bannerId || (banner as APIMiscellaneous).mscId,
  imageUrl: banner.imageUrl
});

const createBanners = ({ data: widgetData }: APIWidget): { banners: Banner[] } => ({
  banners: ((widgetData.banners || widgetData.miscellaneous || []) as (APIBanner | APIMiscellaneous)[])
    .map(apiBannerToBanner)
});

const pivotBanner = (widget: APIWidget) => ({
  ...widgetBase(widget),
  ...createBanners(widget)
});

const createBannersWithTitle = (widget: APIWidget): { banners: BannerWithTitle[] } => ({
  banners: ((widget.data.miscellaneous || [])as APIMiscellaneous[])
    .map(banner => ({
      ...apiBannerToBanner(banner),
      title: banner.name
    }))
});

const createProducts = (widget: APIWidget): { products: DashboardProductPivot[] } => ({
  products: widget.data.products!.map(pivotDashboardProduct)
});

const pivotBannersWithTitle = (widget: APIWidget) => ({
  ...widgetBase(widget),
  ...createBannersWithTitle(widget)
});

const pivotProducts = (widget: APIWidget) => ({
  ...widgetBase(widget),
  ...createProducts(widget)
});

export const pivotDashboardResponse = ({ widgets }: DashboardResponseType) => (
  widgets
    .filter(w => w.data.banners || w.data.miscellaneous || w.data.products)
    .map(widget => {
      switch (widget.type) {
        case 1: return pivotBannersWithTitle(widget) as HorizontalScrollableIcons;
        case 3: return pivotBanner(widget) as Pager;
        case 4: return pivotBanner(widget) as ThinCarouselBanner;
        case 5: return pivotBanner(widget) as BigSquareCarouselBanner;
        case 6:
          return {
            ...widgetBase(widget),
            title: widget.title,
            ...createBanners(widget)
          } as SmallSquareCarouselBanner;
        case 7:
          return {
            title: widget.title,
            viewAllLabel: widget.deepLinkText,
            ...pivotProducts(widget)
          } as HorizontalProducts;
        case 11: return pivotBannersWithTitle(widget) as W4x;
        case 12:
          return {
            title: widget.title,
            ...pivotBannersWithTitle(widget)
          } as W9x;
        case 15: return {
          ...widgetBase(widget),
          title: widget.title,
          ...createBanners(widget)
        } as W4xWithBanner;
        case 16: return {
          ...widgetBase(widget),
          title: widget.title,
          ...createBanners(widget)
        } as W9xWithBanner;
        default: return null;
      }
    })
    .filter(<T>(x: T | null): x is T => x !== null)
);

export const getWidgetDefinitions = promiseMemoize(async () => {
  const { preferences: { pincode, storeId = 0 } } = await getUser();

  const apiWidgets = await makeApiCall<DashboardRequestType, DashboardResponseType>(apiRoutes.getDashboard, {
    body: { pincode, storeId }
  });

  return pivotDashboardResponse(apiWidgets);
}, { resolve: 'json' });
