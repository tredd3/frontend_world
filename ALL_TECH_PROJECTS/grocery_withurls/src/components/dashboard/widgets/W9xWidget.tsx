/** @jsx jsx */
import { jsx, css } from '@emotion/core';
import { Link } from 'react-router-dom';
import Image from '../../uiControls/Image';
import { W9x } from '../../../services/dashboard2';
import { trackLink } from '../../../helpers/analytics';
import linkStyles from '../linkStyles';
import { categoryUrl } from '../../../helpers/urls';

const CONTAINER_PADDING_HORIZONTAL = 16;
const GRID_GAP_HORIZONTAL = 8;
const IMAGE_WIDTH = Math.floor((window.outerWidth - (2 * CONTAINER_PADDING_HORIZONTAL) - (2 * GRID_GAP_HORIZONTAL)) / 3);

const styles = {
  wrapper: css`
    padding: 12px ${CONTAINER_PADDING_HORIZONTAL}px;
    border-top: 5px solid #F2F2F2;
  `,
  grid: css`
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    grid-gap: 15px ${GRID_GAP_HORIZONTAL}px;
    margin-top: 15px;
  `,
  imageWrapper: css`
    border-radius: 3px;
  `,
  title: css`
    color: #333;
    font-size: 13px;
    height: calc(1.46429em * 2);
    overflow: hidden;
    margin-top: 5px;
  `
};

type Props = {
  widgetData: W9x;
};

const W9xWidget = ({
  widgetData: {
    banners,
    title
  }
}: Props) => {
  if (banners.length) {
    return (
      <section css={styles.wrapper} data-testid="Widget12">
        <div css={css`font-weight: 600;`}>{title}</div>
        <div css={styles.grid}>
          {banners.map(({ imageUrl, title, id }, index) => (
            <Link
              data-testid={`banner-link-${index}`}
              key={id}
              css={[linkStyles, css`text-align: center;`]}
              to={categoryUrl(id)}
              onClick={() => {
                trackLink(`Shop By Category | ${title}`, 'HomeScreenClicks', 'Middle');
              }}
            >
              <div css={styles.imageWrapper}>
                <Image width={IMAGE_WIDTH} css={css`border-radius: 3px;`} alt="img" src={imageUrl} />
              </div>
              <div css={styles.title}>{title}</div>
            </Link>
          ))}
        </div>
      </section>
    );
  }

  return null;
};

export default W9xWidget;
