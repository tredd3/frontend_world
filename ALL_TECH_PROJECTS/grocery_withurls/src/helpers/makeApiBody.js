import store from '../store';
import { getUser } from '../services/user';
import { getCart } from '../services/cart';

const getCartId = () => store.getState().cart.CartId || 0;
const getSelectedStoreId = async () => {
  const user = await getUser();
  const { preferences: { storeId } } = user;
  return storeId || undefined;
};

export const getUserId = () => store.getState().user.userId;

export const addCartApiBody = async ({
  skuId, Quantity = 1, ProductType = 0, deliveryModes, addressIdForCheckout
}) => {
  let apiBody = {};

  apiBody = {
    product: [{
      skuId, qty: Number(Quantity), productType: ProductType
    }]
  };

  // Here action is mandatory parameter to be passed. Because based on it backend internally calls checkout api.
  // action : 1 means checkout call
  // action : 0 means no checkout call
  // Also we need to pass other parameters like deliveryDate, timeSlot which needed for checkout api call at backend
  // when action is 1

  if (deliveryModes && Object.keys(deliveryModes).length !== 0) {
    const { date, timeSlot } = deliveryModes;
    apiBody = {
      ...apiBody, action: 1, deliveryDate: date, timeslot: timeSlot, customerAddressId: addressIdForCheckout
    };
  } else {
    apiBody = { ...apiBody, action: 0 };
  }

  const user = await getUser();
  if (user) {
    const { userId } = user;
    apiBody = { ...apiBody, userId };
  }

  const storeId = await getSelectedStoreId();
  if (storeId) {
    apiBody = { ...apiBody, storeId };
  }
  try {
    const cart = await (storeId ? getCart(storeId) : getCart());
    if (cart && cart.cartId !== 0) {
      const { cartId } = cart;
      apiBody = { ...apiBody, cartId };
    }
  } catch (e) {
    console.log(e.message);
  }

  return apiBody;
};

export const removeCartApiBody = ({ skuId, Quantity = 0, ProductType = 0 }) => ({
  cartId: getCartId(),
  skuId,
  qty: Quantity,
  type: ProductType
});

export const moveItemApiBody = ({
  skuId, Quantity = 0, ProductType = 0, moveType
}) => ({
  cartId: getCartId(),
  skuId,
  qty: Quantity,
  productType: ProductType,
  moveType
});

export const removeSaveLaterCartApiBody = ({
  skuId, Quantity = 0, ProductType = 0, moveType
}) => ({
  skuId,
  qty: Quantity,
  productType: ProductType,
  productCategoryType: moveType
});
