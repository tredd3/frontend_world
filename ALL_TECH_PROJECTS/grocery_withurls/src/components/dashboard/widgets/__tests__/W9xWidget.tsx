import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { mocked } from 'ts-jest/utils';
import W9xWidget from '../W9xWidget';
import { pivotDashboardResponse, W9x, DashboardResponseType } from '../../../../services/dashboard2';
import mockDB from '../../../../mocks/db.json';
import * as analytics from '../../../../helpers/analytics';

jest.mock('../../../../helpers/analytics');

const { trackLink } = mocked(analytics);

const mockWidgetData = pivotDashboardResponse(mockDB.dashboard.Data as DashboardResponseType)
  .find(({ type }) => type === 12) as W9x;

describe('W9xWidget tests', () => {
  it('should render the widget correctly', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <W9xWidget
          widgetData={mockWidgetData}
        />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should handle click action correctly for banner links if App interface is not defined', () => {
    const { getByTestId } = render(
      <MemoryRouter>
        <W9xWidget
          widgetData={mockWidgetData}
        />
      </MemoryRouter>
    );

    mockWidgetData.banners.forEach((_banner, index) => {
      fireEvent.click(getByTestId(`banner-link-${index}`));
      expect(trackLink).toHaveBeenCalledTimes(index + 1);
      expect(trackLink.mock.calls[index]).toMatchSnapshot();
    });
  });
});
