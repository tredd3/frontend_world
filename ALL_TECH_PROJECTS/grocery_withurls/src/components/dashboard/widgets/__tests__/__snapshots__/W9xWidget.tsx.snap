// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`W9xWidget tests should handle click action correctly for banner links if App interface is not defined 1`] = `
Array [
  "Shop By Category | Foodgrains, Oils And Masala",
  "HomeScreenClicks",
  "Middle",
]
`;

exports[`W9xWidget tests should handle click action correctly for banner links if App interface is not defined 2`] = `
Array [
  "Shop By Category | Snacks And Confectionery",
  "HomeScreenClicks",
  "Middle",
]
`;

exports[`W9xWidget tests should handle click action correctly for banner links if App interface is not defined 3`] = `
Array [
  "Shop By Category | Health Drinks & Beverages",
  "HomeScreenClicks",
  "Middle",
]
`;

exports[`W9xWidget tests should handle click action correctly for banner links if App interface is not defined 4`] = `
Array [
  "Shop By Category | Beauty & Hygiene",
  "HomeScreenClicks",
  "Middle",
]
`;

exports[`W9xWidget tests should handle click action correctly for banner links if App interface is not defined 5`] = `
Array [
  "Shop By Category | Cleaning & Household Care",
  "HomeScreenClicks",
  "Middle",
]
`;

exports[`W9xWidget tests should handle click action correctly for banner links if App interface is not defined 6`] = `
Array [
  "Shop By Category | Mom & Baby",
  "HomeScreenClicks",
  "Middle",
]
`;

exports[`W9xWidget tests should handle click action correctly for banner links if App interface is not defined 7`] = `
Array [
  "Shop By Category | Dairy, Bakery & Egg",
  "HomeScreenClicks",
  "Middle",
]
`;

exports[`W9xWidget tests should handle click action correctly for banner links if App interface is not defined 8`] = `
Array [
  "Shop By Category | Instant and Ready Foods",
  "HomeScreenClicks",
  "Middle",
]
`;

exports[`W9xWidget tests should render the widget correctly 1`] = `
<DocumentFragment>
  .emotion-34 {
  padding: 12px 16px;
  border-top: 5px solid #F2F2F2;
}

.emotion-0 {
  font-weight: 600;
}

.emotion-33 {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-gap: 15px 8px;
  margin-top: 15px;
}

.emotion-4 {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
  text-align: center;
}

.emotion-4:hover,
.emotion-4:focus,
.emotion-4:visited,
.emotion-4:active {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
}

.emotion-2 {
  border-radius: 3px;
}

.emotion-1 {
  border-radius: 3px;
}

.emotion-3 {
  color: #333;
  font-size: 13px;
  height: calc(1.46429em * 2);
  overflow: hidden;
  margin-top: 5px;
}

<section
    class="emotion-34"
    data-testid="Widget12"
  >
    <div
      class="emotion-0"
    >
      Shop By Category
    </div>
    <div
      class="emotion-33"
    >
      <a
        class="emotion-4"
        data-testid="banner-link-0"
        href="/categories/10001763"
      >
        <div
          class="emotion-2"
        >
          <span
            class=" lazy-load-image-background "
            style="background-image: url(Placeholder.jpg); background-size: 100% 100%; color: transparent; display: inline-block; width: 325px;"
          >
            <img
              alt="img"
              class="emotion-1"
              src="https://jiomerchant.akamaized.net/static/delivery_associate/c2xXAckEA1Oith5HHeuq8MhJdJtEut4PFSE88YhpFA.png?impolicy=resizeforwidth&w=325"
              width="325"
            />
          </span>
        </div>
        <div
          class="emotion-3"
        >
          Foodgrains, Oils And Masala
        </div>
      </a>
      <a
        class="emotion-4"
        data-testid="banner-link-1"
        href="/categories/10001766"
      >
        <div
          class="emotion-2"
        >
          <span
            class=" lazy-load-image-background "
            style="background-image: url(Placeholder.jpg); background-size: 100% 100%; color: transparent; display: inline-block; width: 325px;"
          >
            <img
              alt="img"
              class="emotion-1"
              src="https://jiomerchant.akamaized.net/static/delivery_associate/iIU0LUqS3x+1Kut2wiDAjaFJyyyiO3cSuB0n5lhM9h8.png?impolicy=resizeforwidth&w=325"
              width="325"
            />
          </span>
        </div>
        <div
          class="emotion-3"
        >
          Snacks And Confectionery
        </div>
      </a>
      <a
        class="emotion-4"
        data-testid="banner-link-2"
        href="/categories/10001761"
      >
        <div
          class="emotion-2"
        >
          <span
            class=" lazy-load-image-background "
            style="background-image: url(Placeholder.jpg); background-size: 100% 100%; color: transparent; display: inline-block; width: 325px;"
          >
            <img
              alt="img"
              class="emotion-1"
              src="https://jiomerchant.akamaized.net/static/delivery_associate/5CAzReHV2nAJdYa+V1o78mVubdf9+jH+j4cseEq3K4.png?impolicy=resizeforwidth&w=325"
              width="325"
            />
          </span>
        </div>
        <div
          class="emotion-3"
        >
          Health Drinks & Beverages
        </div>
      </a>
      <a
        class="emotion-4"
        data-testid="banner-link-3"
        href="/categories/10001760"
      >
        <div
          class="emotion-2"
        >
          <span
            class=" lazy-load-image-background "
            style="background-image: url(Placeholder.jpg); background-size: 100% 100%; color: transparent; display: inline-block; width: 325px;"
          >
            <img
              alt="img"
              class="emotion-1"
              src="https://jiomerchant.akamaized.net/static/delivery_associate/4i7mEwBdgChrDntV9sHrCGNF8QbV0wTCogQHa0gJgPE.png?impolicy=resizeforwidth&w=325"
              width="325"
            />
          </span>
        </div>
        <div
          class="emotion-3"
        >
          Beauty & Hygiene
        </div>
      </a>
      <a
        class="emotion-4"
        data-testid="banner-link-4"
        href="/categories/10001762"
      >
        <div
          class="emotion-2"
        >
          <span
            class=" lazy-load-image-background "
            style="background-image: url(Placeholder.jpg); background-size: 100% 100%; color: transparent; display: inline-block; width: 325px;"
          >
            <img
              alt="img"
              class="emotion-1"
              src="https://jiomerchant.akamaized.net/static/delivery_associate/PsdxDGsKudO6A4C+8wX3E3wN7ox7EsF8CCDI+BkJ+s.png?impolicy=resizeforwidth&w=325"
              width="325"
            />
          </span>
        </div>
        <div
          class="emotion-3"
        >
          Cleaning & Household Care
        </div>
      </a>
      <a
        class="emotion-4"
        data-testid="banner-link-5"
        href="/categories/10001764"
      >
        <div
          class="emotion-2"
        >
          <span
            class=" lazy-load-image-background "
            style="background-image: url(Placeholder.jpg); background-size: 100% 100%; color: transparent; display: inline-block; width: 325px;"
          >
            <img
              alt="img"
              class="emotion-1"
              src="https://jiomerchant.akamaized.net/static/delivery_associate/Ny1QvT7RJnIusvdRRYDGB7s8djtawnvfn0r2duIjHU.png?impolicy=resizeforwidth&w=325"
              width="325"
            />
          </span>
        </div>
        <div
          class="emotion-3"
        >
          Mom & Baby
        </div>
      </a>
      <a
        class="emotion-4"
        data-testid="banner-link-6"
        href="/categories/10001767"
      >
        <div
          class="emotion-2"
        >
          <span
            class=" lazy-load-image-background "
            style="background-image: url(Placeholder.jpg); background-size: 100% 100%; color: transparent; display: inline-block; width: 325px;"
          >
            <img
              alt="img"
              class="emotion-1"
              src="https://jiomerchant.akamaized.net/static/delivery_associate/aYVvmNHMVAk5JtQvE8+PR8cT2tBfr2GtHR7r8Pmx1F4.png?impolicy=resizeforwidth&w=325"
              width="325"
            />
          </span>
        </div>
        <div
          class="emotion-3"
        >
          Dairy, Bakery & Egg
        </div>
      </a>
      <a
        class="emotion-4"
        data-testid="banner-link-7"
        href="/categories/10001765"
      >
        <div
          class="emotion-2"
        >
          <span
            class=" lazy-load-image-background "
            style="background-image: url(Placeholder.jpg); background-size: 100% 100%; color: transparent; display: inline-block; width: 325px;"
          >
            <img
              alt="img"
              class="emotion-1"
              src="https://jiomerchant.akamaized.net/static/delivery_associate/V7h3HKP7a5tA4ydP5JT0Q4BgU6DEXEDnN8FQeDHhslw.png?impolicy=resizeforwidth&w=325"
              width="325"
            />
          </span>
        </div>
        <div
          class="emotion-3"
        >
          Instant and Ready Foods
        </div>
      </a>
    </div>
  </section>
</DocumentFragment>
`;
